server/world/block.go

```diff
@@ -75,6 +75,17 @@ type Liquid interface {
 	Harden(pos cube.Pos, w *World, flownIntoBy *cube.Pos) bool
 }
 
+// Conductor represents a block that can conduct a redstone signal.
+type Conductor interface {
+	Block
+	// Source returns true if the conductor is a signal source.
+	Source() bool
```

Comment:
I'd consider renaming this to RedstoneSource or PowerSource for better understanding